core/txpool/legacypool/legacypool.go

```diff
@@ -217,12 +218,13 @@ type LegacyPool struct {
 	locals  *accountSet // Set of local transaction to exempt from eviction rules
 	journal *journal    // Journal of local transaction to back up to disk
 
-	reserve txpool.AddressReserver       // Address reserver to ensure exclusivity across subpools
-	pending map[common.Address]*list     // All currently processable transactions
-	queue   map[common.Address]*list     // Queued but non-processable transactions
-	beats   map[common.Address]time.Time // Last heartbeat from each known account
-	all     *lookup                      // All transactions to allow lookups
-	priced  *pricedList                  // All transactions sorted by price
+	reserve txpool.AddressReserver                  // Address reserver to ensure exclusivity across subpools
+	pending map[common.Address]*list                // All currently processable transactions
+	queue   map[common.Address]*list                // Queued but non-processable transactions
+	beats   map[common.Address]time.Time            // Last heartbeat from each known account
+	all     *lookup                                 // All transactions to allow lookups
+	priced  *pricedList                             // All transactions sorted by price
+	auths   map[common.Address][]*types.Transaction // All accounts with a pooled authorization
```

Comment:
Any particular reason for tracking the tx object itself, rather than the tx hash?

Reply:
It could definitely track the tx hash instead. Is there a preference? I originally went with the point because it is how txs are generally tracked in the pool, but later noticed that we never actually need to use any of values of the tx and it's really just a tracker.

Reply:
tx is larger than txhash

> but later noticed that we never actually need to use any of values of the tx and it's really just a tracker.

Exactly, I would prefer some memory-efficient approach, while the memory consumption is really negligible

Reply:
Ah, right, It's a pointer. Though I feel it's a bit weird to track the entire transaction and the transaction fields are never used.